torchtext/data/datasets_utils.py

```diff
@@ -279,18 +280,39 @@ def _download_extract_validate(root, url, url_md5, downloaded_file, extracted_fi
     return extracted_file
 
 
+class _RawTextMapDataset(torch.utils.data.Dataset):
+    """Helper class to convert IterableDatatset to Dataset"""
+
+    def __init__(self, raw_text_iter_data):
+        """Initialized with _RawTextIterableDataset"""
+        self.description = raw_text_iter_data.description
+        self._data = [d for d in raw_text_iter_data]
```

Comment:
Using ```list(raw_text_iter_data)``` could potentially be more efficient. But large lists of text are difficult in general (e.g. see https://github.com/pytorch/pytorch/issues/13246). As a mitigation we could explore using numpy tensors to store strings or a light custom class that's effectively a wrapper around ```std::vector<std::string>```